cmd/mdatagen/metrics_v2.tmpl

```diff
@@ -0,0 +1,203 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by mdatagen. DO NOT EDIT.
+
+// {{ .ExpFileNote }}
```

Comment:
Don't need this in the template context you can embed the text here.

Reply:
I moved to a const just to use in tests for generated file validation. Just don't like to repeat it there. May we be can keep it?

Reply:
I just want to make sure that the correct file is generated based on the flag, so I'm using this line as anchor. Let me know if you want me to duplicate this string in tests. Or I can generate a file with different name like `genrerate_metrics_v2.go`

Reply:
generate file with different is actually better I think, since that way we may even be able to have both in parallel.

Reply:
Do we need to have both in parallel? I think we should keep only one to make sure the old functionality is not being used be scraper

Reply:
Moved it to `generate_metrics_v2.go` but removed the old one.

Reply:
But now at least it's possible to have both enabled. If we ever need that, we can change the generator command to support that